File: ios/AgenticInbox/Inboxies/Services/DeliveryStatus.swift
import Foundation

/// Outbound delivery lifecycle — mirrors `app/lib/delivery-status.ts`.
enum DeliveryStatusHelpers {
    static func isFailure(_ status: String?) -> Bool {
        guard let status else { return false }
        switch status {
        case "failed", "bounced", "complained":
            return true
        default:
            return false
        }
    }

    static func label(for status: String?) -> String {
        switch status {
        case "failed":
            return "Send failed"
        case "bounced":
            return "Bounced"
        case "complained":
            return "Marked as spam"
        case "queued":
            return "Sending"
        case "accepted":
            return "Sent"
        default:
            return "Delivery issue"
        }
    }
}

File: android/app/src/main/java/co/inboxies/app/util/DeliveryStatus.kt
package co.inboxies.app.util

/** Outbound delivery lifecycle — mirrors `app/lib/delivery-status.ts`. */
object DeliveryStatus {
    fun isFailure(status: String?): Boolean =
        status == "failed" || status == "bounced" || status == "complained"

    fun label(status: String?): String = when (status) {
        "failed" -> "Send failed"
        "bounced" -> "Bounced"
        "complained" -> "Marked as spam"
        "queued" -> "Sending"
        "accepted" -> "Sent"
        else -> "Delivery issue"
    }
}
